Sources/Tests/UnitTests/Core/Settings.cs

```diff
@@ -58,5 +60,29 @@ public void AssertGetBackDespiteDispose()
             }
             Assert.Equal(67, MaxExpansionTermCount.Value);
         }
+
+        [Fact]
+        public void AssertErrorWhenPoweringNumberAccordingToSetting1()
+        {
+            using var _ = ExplicitParsingOnly.Set(true);
+
+            Assert.Throws<InvalidArgumentParseException>(() => {
+                Entity expr1 = "x2=16";
+                expr1.Solve("x");
+            });
+
+        }
+
+        [Fact]
+        public void AssertErrorWhenPoweringNumberAccordingToSetting2()
+        {
+            using var _ = ExplicitParsingOnly.Set(true);
+      
+            Assert.Throws<InvalidArgumentParseException>(() => {
+                Entity expr1 = "x2 + 4x +4 = 0";
+                expr1.Solve("x");
+            });
+
+        }
```

Comment:
Minor issue:
You don't really need `Solve`. But because you can't have a variable initialization as the only line of a lambda expression, you can write `() => MathS.FromString("x2=16")`. I'd actually go with `[Theory]` and `[InlineData]`, so that you can add each new test just by adding one line. You can check examples of using `[Theory]` [here](https://github.com/asc-community/AngouriMath/blob/6f0f859a0b4bb75612ae11b91199268d2b003b42/Sources/Tests/UnitTests/Calculus/IntegrationTest.cs#L10)

Reply:
Okay,that seems much simpler by using [Theory] and [InlineData] from xunit to do test with multiple arguments